crates/rpc/rpc/src/eth/api/traits/transaction.rs

```diff
@@ -56,11 +68,35 @@ pub type StateCacheDB = CacheDB<StateProviderDatabase<StateProviderBox>>;
 /// This implementation follows the behaviour of Geth and disables the basefee check for tracing.
 #[async_trait::async_trait]
 pub trait EthTransactions: Send + Sync {
+    /// Transaction pool with pending transactions. [`TransactionPool::Transaction`] is the
+    /// supported transaction type.
+    type Pool: TransactionPool;
+
     /// Returns a handle for reading data from disk.
     ///
     /// Data access in default (L1) trait method implementations.
     fn provider(&self) -> &impl BlockReaderIdExt;
 
+    /// Returns a handle for reading data from memory.
+    ///
+    /// Data access in default (L1) trait method implementations.
+    fn cache(&self) -> &EthStateCache;
+
+    /// Returns a handle for reading data from pool.
+    ///
+    /// Data access in default (L1) trait method implementations.
+    fn pool(&self) -> &Self::Pool;
+
+    /// Returns a handle for forwarding received raw transactions.
+    ///
+    /// Access to transaction forwarder in default (L1) trait method implementations.
+    fn raw_tx_forwarder(&self) -> &Option<Arc<dyn RawTransactionForwarder>>;
```

Comment:
after my hotfix https://github.com/paradigmxyz/reth/pull/8664 this should be 
```suggestion
    fn raw_tx_forwarder(&self) -> Option<Arc<dyn RawTransactionForwarder>>;
```